GitUI/CommandsDialogs/RevisionDiffControl.cs

```diff
@@ -349,32 +349,36 @@ private ContextMenuSelectionInfo GetSelectionInfo()
                 isAnyItemWorkTree: isAnyWorkTree,
                 isBareRepository: isBareRepository,
                 allFilesExist: allFilesExist,
+                allDirectoriesExist: allDirectoriesExist,
                 allFilesOrUntrackedDirectoriesExist: allFilesOrUntrackedDirectoriesExist,
                 isAnyTracked: isAnyTracked,
                 supportPatches: supportPatches,
                 isDeleted: isDeleted,
                 isAnySubmodule: isAnySubmodule);
             return selectionInfo;
 
-            static (bool allFilesExist, bool allFilesOrUntrackedDirectoriesExist) FileOrUntrackedDirExists(List<FileStatusItem> items, IFullPathResolver fullPathResolver)
+            static (bool allFilesExist, bool allDirectoriesExist, bool allFilesOrUntrackedDirectoriesExist) FileOrUntrackedDirExists(List<FileStatusItem> items, IFullPathResolver fullPathResolver)
             {
                 bool allFilesExist = items.Any();
-                bool allFilesOrUntrackedDirectoriesExist = items.Any();
+                bool allDirectoriesExist = allFilesExist;
+                bool allFilesOrUntrackedDirectoriesExist = allFilesExist;
                 foreach (var item in items)
                 {
                     var path = fullPathResolver.Resolve(item.Item.Name);
                     var fileExists = File.Exists(path);
-                    allFilesExist = allFilesExist && fileExists;
-                    var fileOrUntrackedDirectoryExists = fileExists || (!item.Item.IsTracked && Directory.Exists(path));
-                    allFilesOrUntrackedDirectoriesExist = allFilesOrUntrackedDirectoriesExist && fileOrUntrackedDirectoryExists;
+                    var directoryExists = Directory.Exists(path);
+                    allFilesExist &= fileExists;
+                    allDirectoriesExist &= directoryExists;
+                    var fileOrUntrackedDirectoryExists = fileExists || (!item.Item.IsTracked && allDirectoriesExist);
```

Comment:
Have you changed this intentionally? Or should it be:
```suggestion
                    var fileOrUntrackedDirectoryExists = fileExists || (!item.Item.IsTracked && directoryExists);
```

Reply:
Intentional, to show the menu only if only submodules are shown